ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HivePushdownSnapshotFilterRule.java

```diff
@@ -124,7 +126,8 @@ private boolean setSnapShotId(RexNode op1, RexNode op2) {
       }
 
       RelOptHiveTable hiveTable = (RelOptHiveTable) relOptTable;
-      hiveTable.getHiveTableMD().setVersionIntervalFrom(Long.toString(snapshotId));
+      String snapshotIdText = snapshotId != NULL_SNAPSHOT_ID ? Long.toString(snapshotId) : null;
```

Comment:
why is this needed if eventually, you just map null-> NULL_SNAPSHOT_ID -> null ?

Reply:
got it, thanks! won't it work if we have `NOT(ISNULL($3)) AND <=($3, NULL)`? if not, that's OK with me

Reply:
it is still `null`:
```
<=($3, NULL) -> null
anything AND null -> null
```

Reply:
so there is no lazy evaluation for AND operators? if left is false -> return false?